apps/internal/oauth/ops/authority/authority.go

```diff
@@ -331,13 +332,15 @@ func NewInfoFromAuthorityURI(authorityURI string, validateAuthority bool) (Info,
 		return Info{}, err
 	}
 
+	// u.Host includes the port, if any, which is required for private cloud deployments
 	return Info{
-		Host:                  u.Hostname(),
-		CanonicalAuthorityURI: fmt.Sprintf("https://%v/%v/", u.Hostname(), tenant),
-		AuthorityType:         authorityType,
-		UserRealmURIPrefix:    fmt.Sprintf("https://%v/common/userrealm/", u.Hostname()),
-		ValidateAuthority:     validateAuthority,
-		Tenant:                tenant,
+		Host:                      u.Host,
+		CanonicalAuthorityURI:     fmt.Sprintf("https://%v/%v/", u.Host, tenant),
```

Comment:
Sorry for not noticing this earlier but we should test that clients behave correctly when the given authority includes a port. I think it would be enough to test the simplest AcquireToken method plus silent auth (the authority is used as a cache key), verifying URLs include the port.